### ISSUE TITLE
hyperparameter tuning

### ISSUE THREAD

**Issue by jg10545** [jg10545/decepticon#26]
Going along with #25- assuming we have a reasonably useful performance metric, should we add (or integrate with) any formal hyperparameter tuning tools?

* too many parameters for grid search
* I suppose random search would be easy to implement
* `hyperopt` is designed more around in-memory optimization (like running a bunch of `sklearn` models), but I've shoehorned deep learning problems into it before. It does Bayesian optimization (SMBO) using a tree-based Parzen estimator.
* there's a [keras-tuner](https://github.com/keras-team/keras-tuner) package, though the examples I've seen so far just use random search
